### ISSUE TITLE
Unable to port DISK UNET from Kornia to Inf2. Compilation taking hours with no signs of progress

### ISSUE THREAD

**Issue by kandakji** [aws-neuron/aws-neuron-sdk#1039]
Hi,

I'm trying to port some models from Kornia. I was able to port NetVlad and LightGlue.

When it comes to [Disk](https://kornia.readthedocs.io/en/v0.6.12/feature.html#kornia.feature.DISK), the trace command from torch_neuronx is `Input tensor is not an XLA tensor: LazyFloatType` although I moved the tensors and the model to the xla device.

So, I started experimenting with `torch.jit.trace` the compiler runs but is just stuck at this debug entry:

`2024-11-20T19:01:27Z INFO 454853 [job.Frontend.0]: Executing: <site-packages>/neuronxcc/starfish/bin/hlo2penguin --input /tmp/ubuntu/neuroncc_compile_workdir/ab6ffbbd-bb47-4739-9ef3-fef030126a68/model.MODULE_16216335577045190367+11b4a2df.hlo_module.pb --out-dir ./ --output penguin.py --layers-per-module=1 --partition --coalesce-all-gathers=false --coalesce-reduce-scatters=false --coalesce-all-reduces=false --emit-tensor-level-dropout-ops --emit-tensor-level-rng-ops --expand-batch-norm-training --enable-native-kernel --native-kernel-auto-cast=matmult-to-bf16`

**Comment by fayyadd** [comment]
Thank you for reaching out! To help us investigate this issue, can you please share the neuron versions in your environment ```pip list | grep neuron``` and the steps to reproduce the issue?

**Comment by jluntamazon** [comment]
Currently this model is not well supported due to the use of Billinear upsampling. There is ongoing work to support this operation, but it will require an upcoming release.

*Original Issue Resolution*
While this cannot yet be resolved, below documents the steps used to create a model which is able to be traced correctly (compilation fails).

The original issue should have been resolvable by just tracing the UNet portion of the model for Neuron rather than the included wrapper. One of the reasons this model does not work well with Neuron is the use of variable shaped tensors in the post-processing stages of the model: https://github.com/kornia/kornia/blob/main/kornia/feature/disk/detector.py#L56

To avoid variable shaped tensors, we can usually trace the compute-heavy (static) portion of the model and then allow the post-processing portion of the model to execute on CPU.

If we supported bilinear upsampling, this would be how I would have modified the script:
```python

import torch
from disk import DISK
import torch_neuronx

# load disk model
conf = {
    "max_num_keypoints": 2000,
}
disk_model = DISK(conf)

# Trace Unet
unet_input = torch.ones(1, 3, 1024, 768).mul(0.5)
compiler_args = '--verbose DEBUG --target trn1 --model-type unet-inference --optlevel 1'
unet = torch_neuronx.trace(disk_model.model.unet, unet_input, compiler_args=compiler_args)

# Replace module with traced Neuron module
disk_model.model.unet = unet

# Create final artifact
disk_input = {"image": unet_input}
neuron_disk_model_unet = torch.jit.trace(disk_model, disk_input, strict=False)
```

**Comment by AbigailFernandes** [comment]
Hi @kandakji , 

Support for `ResizeBilinear` will be part of the Neuron 2.23 Release, which should be out by 5/15. You can update the Neuron SDK to be latest and check whether the issue is resolved. Feel free to re-open this ticket if the issue persists. 

Thank you!

**Comment by rohithkrn** [comment]
Closing this issue. Please reopen if you still face the issue with the upcoming release. Thanks
